app/controllers/oauth2/authorization_controller.rb

```diff
@@ -0,0 +1,150 @@
+# NOTE: This class uses Rails views in a non-legacy manner.  This is by design.
+# Do not attempt to Emberize this controller without first studying the
+# potential security effects it may have (namely CSRF)
+#
+# WARNING: IF YOU HAVE NOT READ RFC 6749, DO NOT TOUCH THIS.
+#
+# Throughout this file there are references to RFC 6749 in parentheses.  The
+# numbers separated by dots are the section id from the spec, the number after
+# "P" is the paragraph, and "S" is sentence.
+class OAuth2::AuthorizationController < ApplicationController
+  # Valid response errors (4.1.2.1, 4.2.2.1)
+  VALID_ERRORS = %i[invalid_request unauthorized_client access_denied
+                    unsupported_response_type invalid_scope server_error
+                    temporarily_unavailable]
+  # Not specified by RFC 6749, these are internal
+  VALID_FATAL_ERRORS = %i[invalid_client invalid_redirect]
```

Comment:
Many of the private methods are just shortcuts for things in params or headers, so those can't be moved.  The validations are also (a) too complex to express via `ActiveModel::Validators` and (b) the entirety of this controller (literally it's just validating and one line of "oh right we should go render that page"

Due to the lack of overlap in implementation between this and the `OAuth2::TokenController`, I don't think it's worthwhile right now to refactor.